### PR TITLE
Source Amazon Ads: get rid of `fail_on_extra_columns: false` in SAT

### PR DESCRIPTION
**Pull request by roman-yermilov-gl**
## What
Amazon Ads passes CAT without declaring `fail_on_extra_columns: false`

## How
The following descriptions of streams that pass undeclared columns come from results of the failed connector acceptance test:

```
columns.txt:./source-amazon-ads.txt-88-The sponsored_display_ad_groups stream has the following schema errors:
columns.txt:./source-amazon-ads.txt:89:Additional properties are not allowed ('creativeType' was unexpected)
columns.txt:./source-amazon-ads.txt-112-The sponsored_display_targetings stream has the following schema errors:
columns.txt:./source-amazon-ads.txt:113:Additional properties are not allowed ('expression', 'resolvedExpression' were unexpected)
columns.txt:./source-amazon-ads.txt-133-The sponsored_product_campaigns stream has the following schema errors:
columns.txt:./source-amazon-ads.txt:134:Additional properties are not allowed ('ruleBasedBudget' was unexpected)
columns.txt:./source-amazon-ads.txt-175-The sponsored_product_targetings stream has the following schema errors:
columns.txt:./source-amazon-ads.txt:176:Additional properties are not allowed ('expression', 'resolvedExpression' were unexpected)

```

1. Add the missing properties indicated by the `Additional properties are not allowed ('<column>', 'column' were unexpected)` logs to the connector's spec.
2. Remove `fail_on_extra_columns: false` from the `acceptance-test-config.yml` file.
3. Commit changes to spec and `acceptance-test-config.yml` and open a PR.
4. Run tests on the connector, either automatically via CI or manually via the `/test` command
5. Profit!

### CODE REVIEW

File: airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/schemas/common.py [hunk 1]
```diff
@@ -36,6 +36,9 @@ def schema_extra(cls, schema: Dict[str, Any], model: Type["BaseModel"]) -> None:
                 if "type" in prop:
                     if allow_none:
                         prop["type"] = ["null", prop["type"]]
+                    if prop["type"] == "array" and prop["items"]:
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/schemas/common.py:39]: let's add unit tests to check this functionality.

**Review by lazebnyi — APPROVED** [review]
LGTM! Minor changes needed.

**Comment by erohmensing** [comment]
@roman-yermilov-gl can you please lay out what changes are made and why?

File: airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/schemas/common.py (hunk 1 above)
Comment (roman-yermilov-gl) [reply airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/schemas/common.py:39]: done

**Pull request merged**